conda/scripts/activate_env_vars.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -x
+
+### Path to clang++ ###
```

Comment:
IIUC this file will be run when we conda activate taichi-dev environment? (great idea!)
Nit: maybe it's good to do some additional checks here? For example we know on linux if CXX is not set taichi will likely pick up g++ instead of clang++, which is not what we want? 
Similar for llvm, IIUC we either need llvm to be in PATH or set LLVM_DIR? we can probably check those here and give nicer errors than at cmake :D 
Same for LLVM_DIR, maybe we can check `llvm-config --version` here to make sure it's in path?

Reply:
Good point... Not exactly sure what additional checks we need here. I feel like CMake is still better at catching such errors, given that it handles these env settings in a more structured way. (E.g. we don't have to write some fancy script just to figure out where LLVM headers/libs are) WDYT?

From my point of view, the bigger issue is that users who change this file would need to remember to reset it, or their working tree becomes dirty...

Reply:
Yea originally I was thinking about some simple checks like "check CXX must be set on linux/llvm-config must be runnable or LLVM_DIR is set".(cmake errors are clear enough but it doesn't remind setting CXX/LLVM_DIR as workarounds :P )  But yea I agree it's kinda hard to remember updating this file compared to normal bashrc/zshrc I guess. I'm totally fine that we start with this version first :D